game/src/game/components/VehicleComponent.cpp

```diff
@@ -158,62 +252,92 @@ void VehicleComponent::HandleVehicleTransform()
     }
 }
 
-void VehicleComponent::OnInit(const ServiceProvider& service_provider)
+void VehicleComponent::UpdateGrounded()
 {
-    physics_service_ = &service_provider.GetService<PhysicsService>();
-    input_service_ = &service_provider.GetService<InputService>();
-    transform_ = &GetEntity().GetComponent<Transform>();
-    game_state_service_ = &service_provider.GetService<GameStateService>();
-    audio_emitter_ = &GetEntity().GetComponent<AudioEmitter>();
+    // get orientation
+    glm::vec3 position = transform_->GetPosition();
+    glm::vec3 down = -transform_->GetUpDirection();
 
-    GetEventBus().Subscribe<OnUpdateEvent>(this);
-    GetEventBus().Subscribe<OnPhysicsUpdateEvent>(this);
+    // check
+    auto raycast_data = physics_service_->RaycastStatic(position, down, 10.0f);
+    is_grounded_ = (raycast_data.has_value()) ? true : false;
```

Comment:
Doesn't `has_value()` already return a bool? if so, couldn't we do:
```c++
is_grounded_ = raycast_data.has_value();
```

Reply:
THAT is quite a GOOD POINT lmao, i am losing my mind